src/rtc.rs

```diff
@@ -0,0 +1,484 @@
+/*!  Interface to the real time clock */
+
+use crate::rcc::{APB1, BDCR, CR, CSR};
+use crate::stm32::{PWR, RTC};
+use rtcc::{Datelike, Hours, NaiveDate, NaiveDateTime, NaiveTime, Rtcc, Timelike};
+
+/// Invalid input error
+// #[derive(Clone, Copy)]
+#[derive(Debug)]
+pub enum Error {
+    InvalidInputData,
+}
+
+/// RTC clock input source
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum RTCSrc {
```

Comment:
Check out this RFC for naming conventions in Rust: https://github.com/rust-lang/rfcs/blob/master/text/0430-finalizing-naming-conventions.md

Notably acronyms count as one word for the CamelCase notation, so did should be `RtcSrc`. But there is no need to obfuscate the name just to save three characters here, so `RtcSource` would be what I prefer.